tests/unit/s2n_server_hello_retry_test.c

```diff
@@ -37,8 +39,56 @@ const uint8_t COMPRESSION_METHOD_SIZE = 1;
 struct client_hello_context {
     int invocations;
     s2n_client_hello_cb_mode mode;
+    bool mark_done;
+    bool enable_poll;
 };
 
+int s2n_negotiate_poll_hello_retry(struct s2n_connection *server_conn,
+                struct s2n_connection *client_conn,
+                struct client_hello_context *client_hello_ctx)
+{
+    bool server_done = false, client_done = false;
+    s2n_blocked_status blocked = S2N_NOT_BLOCKED;
+    bool rc = false;
+    EXPECT_FAILURE_WITH_ERRNO(s2n_negotiate(client_conn, &blocked), S2N_ERR_IO_BLOCKED);
+
+    int expected_invocation = 0;
+
+    /* if polling is enabled then confirm that the callback is incremented each time */
+    if (client_hello_ctx->enable_poll) {
+        /* invocation should increase each time s2n_negotiate is called */
+        EXPECT_FAILURE_WITH_ERRNO(s2n_negotiate(server_conn, &blocked), S2N_ERR_ASYNC_BLOCKED);
+        EXPECT_EQUAL(blocked, S2N_BLOCKED_ON_APPLICATION_INPUT);
+        expected_invocation++;
```

Comment:
Any reason to manually type this out twice for two calls, or could you put it in a loop?